pupa/importers/organizations.py

```diff
@@ -13,30 +15,49 @@ class OrganizationImporter(BaseImporter):
                       'other_names': (OrganizationName, 'organization_id', {}),
                       'contact_details': (OrganizationContactDetail, 'organization_id', {}),
                       'links': (OrganizationLink, 'organization_id', {}),
-                      'sources': (OrganizationSource, 'organization_id', {}),
-                     }
+                      'sources': (OrganizationSource, 'organization_id', {})
+                      }
 
     def get_object(self, org):
-        spec = {'classification': org['classification'],
-                'name': org['name'],
-                'parent_id': org['parent_id']}
-
-        # add jurisdiction_id unless this is a party
-        jid = org.get('jurisdiction_id')
-        if jid:
-            spec['jurisdiction_id'] = jid
-
-        return self.model_class.objects.get(**spec)
+        all_names = [org['name']] + [o['name'] for o in org['other_names']]
+
+        main_query = (Q(name__in=all_names) | Q(other_names__name__in=all_names))
+
+        if org['source_identified']:
```

Comment:
on `Person` and `Organization` classes, `source_identified` defaults to `False` if not otherwise specified, so the tests that are already in place should cover the `else` case, here.

i can add tests to cover the `source_identified == True` case.